src/gluonts/model/seasonal_naive/_predictor.py

```diff
@@ -70,13 +70,13 @@ def __init__(
             else get_seasonality(freq)
         )
 
-    def predict(self, dataset: Dataset, **kwargs) -> Iterator[SampleForecast]:
-        for data in dataset:
-            start = pd.Timestamp(data["start"], freq=self.freq)
-            target = np.asarray(data["target"], np.float32)
-            yield self._predict_time_series(start_time=start, target=target)
+    def predict_item(self, item: DataEntry) -> Forecast:
+        start_time = pd.Timestamp(item["start"], freq=self.freq)
+        target = np.asarray(item["target"], np.float32)
+        return self.predict_time_series(start_time, target)
 
-    def _predict_time_series(
+    # Needed for OWA metric
+    def predict_time_series(
```

Comment:
The SeasonalNaivePredictor was neglected I think, because it didn't correctly implement the RepresentablePredictor functions.